.github/workflows/workflow-common.yml

```diff
@@ -61,18 +61,61 @@ jobs:
           docker system prune -af --volumes
           ./run.py -u
 
+      - name: Archive unit-tests results
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: unit-test-report
+          path: build/test_doc
+  integration_tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
```

Comment:
One thing to consider would be if we actually want to fail-fast. We could have faster feedback in case of errors

Reply:
RIght now we have small amount of tests, so you are right if one of them fail we can notice it faster. But in case of increment number of tests, it would be good to have results of all of them at once, so user don't need to run tests in iterations after fixing one bug at the time, and wait again for executing previously passing. What do you think?

Reply:
Ok makes sense